nighthawk/source/client/benchmark_client_impl.cc

```diff
@@ -0,0 +1,230 @@
+      [this, &dns_resolved, &active_dns_query_](
+          const std::list<Envoy::Network::Address::InstanceConstSharedPtr>&& address_list) -> void {
+        active_dns_query_ = nullptr;
+        ENVOY_LOG(debug, "DNS resolution complete for {} ({} entries).", uri_->host_without_port(),
+                  address_list.size());
+        if (!address_list.empty()) {
+          target_address_ =
+              Envoy::Network::Utility::getAddressWithPort(*address_list.front(), uri_->port());
+          dns_resolved = true;
+        } else {
+          ENVOY_LOG(critical, "Could not resolve host [{}]", uri_->host_without_port());
+        }
+        dispatcher_.exit();
+      });
+  // Wait for DNS resolution to complete before proceeding.
+  dispatcher_.run(Envoy::Event::Dispatcher::RunType::Block);
+  return dns_resolved;
+}
+
+bool BenchmarkClientHttpImpl::initialize(Envoy::Runtime::Loader& runtime) {
+  if (!syncResolveDns()) {
+    return false;
+  }
+
+  envoy::api::v2::Cluster cluster_config;
+  envoy::api::v2::core::BindConfig bind_config;
+
+  cluster_config.mutable_connect_timeout()->set_seconds(timeout_.count());
+  auto thresholds = cluster_config.mutable_circuit_breakers()->add_thresholds();
+
+  thresholds->mutable_max_retries()->set_value(0);
+  thresholds->mutable_max_connections()->set_value(connection_limit_);
+  thresholds->mutable_max_pending_requests()->set_value(max_pending_requests_);
+
+  Envoy::Network::TransportSocketFactoryPtr socket_factory;
+
+  if (uri_->scheme() == "https") {
+    auto common_tls_context = cluster_config.mutable_tls_context()->mutable_common_tls_context();
+    if (use_h2_) {
+      common_tls_context->add_alpn_protocols("h2");
+    }
+    common_tls_context->add_alpn_protocols("http/1.1");
+
+    //
+    auto transport_socket = cluster_config.transport_socket();
+    if (!cluster_config.has_transport_socket()) {
+      ASSERT(cluster_config.has_tls_context());
+      transport_socket.set_name(
+          Envoy::Extensions::TransportSockets::TransportSocketNames::get().Tls);
+      Envoy::MessageUtil::jsonConvert(cluster_config.tls_context(),
+                                      *transport_socket.mutable_config());
+    }
+
+    auto& config_factory = Envoy::Config::Utility::getAndCheckFactory<
+        Envoy::Server::Configuration::UpstreamTransportSocketConfigFactory>(
+        transport_socket.name());
+    Envoy::ProtobufTypes::MessagePtr message =
+        Envoy::Config::Utility::translateToFactoryConfig(transport_socket, config_factory);
+
+    ssl_context_manager_.reset(
+        new Envoy::Extensions::TransportSockets::Tls::ContextManagerImpl(api_.timeSource()));
+    transport_socket_factory_context_ = std::make_unique<Ssl::MinimalTransportSocketFactoryContext>(
+        store_->createScope("client."), dispatcher_, generator_, *store_, api_,
+        *ssl_context_manager_);
+
+    // TODO(oschaaf): We perform some bootstrapping ourselves here, to avoid an assert during the
```

Comment:
What do you mean by bootstrapping?

Reply:
Updated this comment about bootstrapping